### PR TITLE
chore: move /proto into /master

### PR DESCRIPTION
**Pull request by NicholasBlaskey**
## Description

First step of moving us towards a single ```/cluster``` folder that contains a single ```go.mod``` file

## Test Plan

CI passes

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Review by dannysauer — APPROVED** [review]
Infra-relevant stuff here LGTM.

File: master/Makefile [hunk 1]
```diff
@@ -221,3 +230,70 @@ publish-nvcr:
 		docker tag $(DOCKER_REPO)/$(PROJECT_NAME):$(FULL_COMMIT)-amd64 $$image; \
 		docker push $$image; \
 	done
+
+# Proto stuff
+# String constants.
```
Comment (rb-determined-ai) [inline master/Makefile:235]: I would suggest putting the proto stuff first, since it is a more fundamental dependency

File: master/Makefile [hunk 2]
```diff
@@ -98,6 +100,7 @@ build: gen
 		./cmd/determined-master
 
 .PHONY: build-race
+build-race: build-proto
```
Comment (rb-determined-ai) [inline master/Makefile:103]: I don't know if this syntax is correct or not, but I think it definitely makes more sense to put the target dependency on the last line, with the `gen` dependency.

```
build-race: export WHATEVER=
build-race: export WHOCARES=
build-race gen build-proto     # dependent targets go on this line
```

File: go.work [hunk 3]
```diff
@@ -3,5 +3,4 @@ go 1.21
 use (
 	./agent
 	./master
-	./proto
```
Comment (rb-determined-ai) [inline go.work:6]: :tada:

File: master/get-deps.sh [hunk 4]
```diff
@@ -6,3 +6,10 @@ go install github.com/goreleaser/goreleaser@v1.14.1
 go install github.com/swaggo/swag/cmd/swag@v1.8.9
 go install github.com/vektra/mockery/v2@v2.20.0
 go install gotest.tools/gotestsum@v1.9.0
+
+# Proto deps
```
Comment (rb-determined-ai) [inline master/get-deps.sh:10]: again, I recommend putting proto first

**Review by rb-determined-ai — APPROVED** [review]

**Comment by NicholasBlaskey** [comment]
taking an alternate approach https://github.com/determined-ai/determined/pull/8285

**Pull request closed without merging**